doc/source/train/api/deprecated.rst

```diff
@@ -0,0 +1,197 @@
+
+    This page contains the deprecated V1 API references. See :ref:`train-api` for the new V2 API references.
+
+
+PyTorch Ecosystem
+-----------------
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.torch.torch_trainer.TorchTrainer
+    ~train.torch.TorchConfig
+    ~train.torch.xla.TorchXLAConfig
+
+PyTorch
+~~~~~~~
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.torch.get_device
+    ~train.torch.get_devices
+    ~train.torch.prepare_model
+    ~train.torch.prepare_data_loader
+    ~train.torch.enable_reproducibility
+
+PyTorch Lightning
+~~~~~~~~~~~~~~~~~
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.lightning.prepare_trainer
+    ~train.lightning.RayLightningEnvironment
+    ~train.lightning.RayDDPStrategy
+    ~train.lightning.RayFSDPStrategy
+    ~train.lightning.RayDeepSpeedStrategy
+    ~train.lightning.RayTrainReportCallback
+
+Hugging Face Transformers
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.huggingface.transformers.prepare_trainer
+    ~train.huggingface.transformers.RayTrainReportCallback
+
+
+More Frameworks
+---------------
+
+Tensorflow/Keras
+~~~~~~~~~~~~~~~~
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.tensorflow.tensorflow_trainer.TensorflowTrainer
+    ~train.tensorflow.TensorflowConfig
+    ~train.tensorflow.prepare_dataset_shard
+    ~train.tensorflow.keras.ReportCheckpointCallback
+
+Horovod
+~~~~~~~
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.horovod.HorovodTrainer
+    ~train.horovod.HorovodConfig
+
+
+XGBoost
+~~~~~~~
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.xgboost.xgboost_trainer.XGBoostTrainer
+    ~train.xgboost.RayTrainReportCallback
+
+
+LightGBM
+~~~~~~~~
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.lightgbm.lightgbm_trainer.LightGBMTrainer
+    ~train.lightgbm.RayTrainReportCallback
+
+
+Ray Train Configuration
+-----------------------
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~air.config.ScalingConfig
+    ~air.config.RunConfig
+    ~air.config.FailureConfig
+    ~train.CheckpointConfig
+    ~train.DataConfig
+    ~train.SyncConfig
+
+
+Ray Train Utilities
+-------------------
+
+**Classes**
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.Checkpoint
+    ~train.context.TrainContext
+
+**Functions**
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train._internal.session.get_checkpoint
+    ~train.context.get_context
+    ~train._internal.session.get_dataset_shard
+    ~train._internal.session.report
+
+
+Ray Train Output
+----------------
+
+.. autosummary::
+    :nosignatures:
+    :toctree: doc/
+
+    ~train.Result
```

Comment:
This points to the V2 Result, should we update this to the V1 path? Same for any other classes that are overridden in `train/__init__`.

Reply:
I tried pointing to the V1 Result but I couldn't get the doc build to work. And I didn't think it was worth the effort since it's basically the same thing w/ or w/o some deprecation messages in the __doc__